File: JDK11/opentraceability/src/main/java/opentraceability/models/events/EventDestinationType.java
package opentraceability.models.events;

import opentraceability.utility.CBVAttribute;

public enum EventDestinationType {
    Unknown(0),

    @CBVAttribute(value = "owning_party")
    @CBVAttribute(value = "https://ref.gs1.org/cbv/SDT-owning_party")
    @CBVAttribute(value = "urn:epcglobal:cbv:sdt:owning_party")
    Owner(1),

    @CBVAttribute(value = "possessing_party")
    @CBVAttribute(value = "https://ref.gs1.org/cbv/SDT-possessing_party")
    @CBVAttribute(value = "urn:epcglobal:cbv:sdt:possessing_party")
    Possessor(2),

    @CBVAttribute(value = "location")
    @CBVAttribute(value = "https://ref.gs1.org/cbv/SDT-location")
    @CBVAttribute(value = "urn:epcglobal:cbv:sdt:location")
    Location(3);

    private int value;

    EventDestinationType(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }
}

File: Java/OpenTraceability/src/main/opentraceability/models/events/eventdestinationtype.kt
package models.events
enum class EventDestinationType(val value: Int) {
    //TODO: review this

    Unknown(0),

    //[CBV("owning_party")
    //[CBV("https://ref.gs1.org/cbv/SDT-owning_party")
    //[CBV("urn:epcglobal:cbv:sdt:owning_party")
    Owner(1),

    //[CBV("possessing_party")
    //[CBV("https://ref.gs1.org/cbv/SDT-possessing_party")
    //[CBV("urn:epcglobal:cbv:sdt:possessing_party")
    Possessor(2),

    //[CBV("location")
    //[CBV("https://ref.gs1.org/cbv/SDT-location")
    //[CBV("urn:epcglobal:cbv:sdt:location")
    Location(3),
}
